src/librbd/cache/pwl/Request.cc

```diff
@@ -41,10 +41,7 @@ std::ostream &operator<<(std::ostream &os,
       << "user_req=" << req.user_req << ", "
       << "m_user_req_completed=" << req.m_user_req_completed << ", "
       << "m_deferred=" << req.m_deferred << ", "
-      << "detained=" << req.detained << ", "
-      << "waited_lanes=" << req.waited_lanes << ", "
-      << "waited_entries=" << req.waited_entries << ", "
-      << "waited_buffers=" << req.waited_buffers << "";
+      << "detained=" << req.detained << "";
```

Comment:
I printed the log to verify. When using `std::ostream` output, we cannot guarantee whether there are spaces before and after the current output statement. So in the whole PWL code, some places have added tail spaces, some places have added head spaces, and some places have forgotten to add. Where do you think it should be added uniformly? Tail? I will make further modifications. For example:
```
2021-12-29T16:10:12.944+0800 7f7651acc700 20 librbd::cache::pwl::Request: 0x56188c62fce0 setup_log_operations: write_req=image_extents=[[990789632,16384]], image_extents_summary=[total_bytes=16384, first_image_byte=990789632, last_image_byte=990806016], bl=buffer::list(len=16384,
	buffer::ptr(0~16384 0x56188e5b4000 in raw 0x56188e5b4000 len 16384 nref 1)
), user_req=0x56188bea1290, m_user_req_completed=0, m_deferred=0, detained=0 m_resources.allocated=1op_set=cell=0x56188d08e7e0, extent_ops_appending=[0x56188c824780, extent_ops_persist=[0x56188bd5f2c0] op_set=0x5618839fa480 operation=0x56188c6f8250

```

here `m_resources.allocated=1op_set=cell=0x56188d08e7e0` lose `" "`.

Reply:
I think head spaces are better.  Presumably you always know where you are starting a new line, in all other places just start with a space.

But my comment wasn't about (potentially missing) spaces -- the line I commented on is appending `""` (i.e. an empty string), not a space.

Reply:
I removed it. The original author may want to add a space or `"."`.  Now I add spaces at the head of most `osteam` output, so the tail spaces are no longer needed.  To avoid this form of output: `write_req=image_extents=[[]]`, I referenced other codes and added `[]`. It will show `req=[]`. Now the request ouput shows:

```
2021-12-31T15:50:14.496+0800 7f59f033f700 20 librbd::cache::pwl::Request: 0x5615abb946e0 setup_log_operations: write_req=[image_extents=[64749568,262144], image_extents_summary=[total_bytes=262144, first_image_byte=64749568, last_image_byte=65011712], bl=buffer::list(len=262144,                                                                                                                               
  buffer::ptr(0~262144 0x5615b46de000 in raw 0x5615b46de000 len 262144 nref 1)
), user_req=0x5615abb3e530, m_user_req_completed=0, m_deferred=0, detained=0, waited_lanes=0, waited_entries=0, waited_buffers=0 m_resources.allocated=1 op_set=cell=0x7f59fc98b0e0, extent_ops_appending=[0x5615abb7fb00, extent_ops_persist=[0x5615abb7f800]] op_set=0x5615abb5aa80

```